config/routes.rb

```diff
@@ -536,6 +536,10 @@ def matches?(request)
     post 'recover', on: :member, defaults: { format: 'json' }
   end
 
+  concern :upsertable do
+    put '', on: :collection, defaults: { format: 'json' }, action: :create_or_update
```

Comment:
Does the following work?

```suggestion
    put on: :collection, defaults: { format: 'json' }, action: :create_or_update
```

Reply:
nah, I get "Route path not specified" if path.nil